### PR TITLE
Expect GPUOutOfMemoryError in map_oom tests

### LINKED ISSUE #199: webgpu:api,operation,buffers,map_oom: only checks for a JS exception
This test checks for a JS exception, expecting that mapping an enormous range would cause a RangeError due to failure to allocate an ArrayBuffer that large. On some platforms, this allocation actually succeeds, and then the test fails with an OOM error when the implementation attempts to map such a large GPU buffer.

I suggest this test be updated to allow a GPU OOM error if there is no JS exception.

### PR DESCRIPTION
**Pull request by austinEng**
Fixes #199. Also adds a helper to asynchronously expect errors.

### CODE REVIEW

**Comment by austinEng** [comment]
I believe this will probably fail on Windows NVIDIA in Chrome Canary (because it doesn't actually OOM) somehow, though I haven't been able to repro locally on my Windows NVIDIA workstation. Probably a Dawn/Chrome bug though.

**Comment by austinEng** [comment]
I realized I should make this against `master`. I'll rebase.

File: src/webgpu/gpu_test.ts [hunk 1]
```diff
@@ -226,4 +226,40 @@ got [${failedByteActualValues.join(', ')}]`;
     fillTextureDataWithTexelValue(expectedTexelData, format, dimension, arrayBuffer, size, layout);
     this.expectContents(buffer, new Uint8Array(arrayBuffer));
   }
+
+  expectError<R>(filter: GPUErrorFilter, fn: () => R): R {
```
Comment (kainino0x) [inline src/webgpu/gpu_test.ts:230]: nit: perhaps `expectGPUError`?

File: src/webgpu/api/operation/buffers/map_oom.spec.ts [hunk 2]
```diff
@@ -3,27 +3,31 @@ export const description = '';
...
 }
 
 export const g = makeTestGroup(GPUTest);
 
 g.test('mapWriteAsync').fn(async t => {
-  const buffer = t.device.createBuffer(getBufferDesc());
+  const buffer = t.expectError('out-of-memory', () => {
+    return t.device.createBuffer(getBufferDesc(GPUBufferUsage.MAP_WRITE));
+  });
   t.shouldReject('RangeError', buffer.mapWriteAsync());
```
Comment (kainino0x) [inline src/webgpu/api/operation/buffers/map_oom.spec.ts:19]: I think these should be DOMExceptions (OperationErrors) now.

File: src/webgpu/api/operation/buffers/map_oom.spec.ts [hunk 3]
```diff
@@ -3,27 +3,31 @@ export const description = '';
...
 g.test('mapReadAsync').fn(async t => {
-  const buffer = t.device.createBuffer(getBufferDesc());
+  const buffer = t.expectError('out-of-memory', () => {
+    return t.device.createBuffer(getBufferDesc(GPUBufferUsage.MAP_READ));
+  });
   t.shouldReject('RangeError', buffer.mapReadAsync());
 });
 
 g.test('createBufferMapped').fn(async t => {
   t.shouldThrow('RangeError', () => {
```
Comment (kainino0x) [inline src/webgpu/api/operation/buffers/map_oom.spec.ts:30]: (should still be RangeError probably)

**Comment by kainino0x** [comment]
> I realized I should make this against `master`. I'll rebase.

Huh. I didn't realize you could change which branch a PR is against.

File: src/webgpu/gpu_test.ts (hunk 1 above)
Comment (austinEng) [reply src/webgpu/gpu_test.ts]: Done

**Comment by austinEng** [comment]
> Huh. I didn't realize you could change which branch a PR is against.

I didn't either; maybe it's a new Github thing.

**Pull request merged**